### PR TITLE
Introduce autoNANO

### PR DESCRIPTION
**Pull request by simonepigazzini**
#### PR description:

Introduce a new utility to configure the NANO step using a mapping technique similar to the one used by DQM to load only selected modules.

autoNANO allows one to customize the NANO step using a syntax like: `-s NANO:@PHYS+@L1Full` where the `@` is then mapped to sequences and customize function and loaded into the processing. Previously we used the standard  `-s NANO:my_cff.my_sq1+sq2` syntax to customize the NANO step, this is still supported, but the two syntaxes cannot be mixed.

We also introduce a different mapping function than the one used by DQM, mainly because the latter uses a list of lists to specify different modules (pre, main, post) while we prefer to use a dictionary to avoid ordering related bugs (we can discuss if we actually want to port the DQM implementation to use a dictionary instead of a list of lists as well).

In this PR we insert a minimal set of workflows into autoNANO, the list will be expanded soon collecting the other existing workflow that are currently configured in a different way.

#### PR validation:

Ran few test jobs for all possible combination. Successfully  reproduced PromptNano 2023 and standard NANO ntuples.

No backport foreseen at the moment.

#### Further details on the autoNANO syntax

The default NANO sequence (which includes a "default customize function") is loaded by default if no step modifer is specified (`-s NANO`) or if the mapping points to empty strings (`'PHYS': {'sequence': '',  'customize': ''},). Missing dictionary keys are instead used to specify a workflow that do not require a dedicated "sequence" or "customize".

### CODE REVIEW

**Comment by vlimant** [comment]
please test

**Comment by simonepigazzini** [comment]
please test

**Comment by simonepigazzini** [comment]
+1

**Comment by antoniovilela** [comment]
+1

File: Configuration/Applications/python/ConfigBuilder.py [hunk 1]
```diff
@@ -1776,9 +1777,30 @@ def prepare_NANO(self, stepSpec = '' ):
         print(f"in prepare_nano {stepSpec}")
         ''' Enrich the schedule with NANO '''
         _,_nanoSeq,_nanoCff = self.loadDefaultOrSpecifiedCFF(stepSpec,self.NANODefaultCFF,self.NANODefaultSeq)
-        self.scheduleSequence(_nanoSeq,'nanoAOD_step')
-        custom = "nanoAOD_customizeCommon"
-        self._options.customisation_file.insert(0,'.'.join([_nanoCff,custom]))
```
Comment (vlimant) [inline Configuration/Applications/python/ConfigBuilder.py:1781]: NB : inserting at 0, versus appending has undesirable effects. #44908 helps once all is in autoNano

**Pull request merged**